internal/datacoord/meta.go

```diff
@@ -51,24 +53,21 @@ import (
 
 type meta struct {
 	sync.RWMutex
-	ctx            context.Context
-	catalog        metastore.DataCoordCatalog
-	collections    map[UniqueID]*collectionInfo // collection id to collection info
-	segments       *SegmentsInfo                // segment id to segment info
+	ctx     context.Context
+	catalog metastore.DataCoordCatalog
+	// TODO there is a leakage on collections, we are not cleanup collectionInfo under drop
+	collections map[UniqueID]*collectionInfo // collection id to collection info
+	// guard to avoid concurrent write on segments, key is segmentID,
+	segmentLock *lock.KeyLock[string]
```

Comment:
If use `*lock.KeyLock[typeutil.UniqueID]` here, we don't need to do additional string conversion when asking for lock.

Reply:
> If use `*lock.KeyLock[typeutil.UniqueID]` here, we don't need to do additional string conversion when asking for lock.

Gotch, will fix it soon